qiskit_experiments/data_processing/processor_library.py

```diff
@@ -98,3 +116,66 @@ def get_processor(
         return processor
 
     raise DataProcessorError(f"Unsupported measurement level {meas_level}.")
+
+
+def is_restless(experiment_data: ExperimentData, analysis_options: Options, index: -1) -> bool:
+    """Check if the experiment data and analysis options correspond to a restless experiment.
+    An experiment is run in a restless setting if the qubits are not reset to the ground
+    state after the measurement and the repetition delay is set to a value much smaller
+    than the T1 value of the physical qubits in the experiment.
+
+    Args:
+        experiment_data: The experiment data that holds all the data and metadata needed
+            to determine the data processor to use to process the data for analysis.
+        analysis_options: The analysis options with which to analyze the data.
+        index: The index of the job for which to get a data processor. The default value is -1.
+
+    Returns:
+        True if the experiment is classified as a restless experiment.
+
+    Notes:
+        The following relevant arguments are extracted from the experiment_data metadata run options:
+            - init_qubits (bool): If False, the qubits are not reset to the ground state after a
+              measurement. The default is True.
+            - physical qubits: The physical qubits used in the experiment.
+            - rep_delay (float): The delay between a measurement and the subsequent circuit.
+              The default is None.
+
+        In addition, the following argument is extracted from the experiment_data:
+            - t1_values (List[float]): The T1 values of the physical qubits at the time
+              of the experiment.
+
+    Raises:
+        DataProcessorError: if excited state promotion readout is enabled in a restless setting.
+        DataProcessorError: if the experiment is run with more than on qubit.
+    """
+    run_options = experiment_data.metadata["job_metadata"][index].get("run_options", {})
+    init_qubits = run_options.get("init_qubits", True)
+    rep_delay = run_options.get("rep_delay", None)
+    restless_threshold = analysis_options.get("restless_threshold", 1)
+    esp_enabled = analysis_options.get("use_measure_esp", False)
+
+    if rep_delay and not init_qubits:
```

Comment:
I think it would be good to provide an option like `use_restless` to short circuit this logic by explicitly enabling or disabling restless processing. One could pass a processor explicitly but that argument applies to the need for an `is_restless` function as well.